packages/popper/src/index.vue

```diff
@@ -0,0 +1,409 @@
+    },
+    hideAfter: {
+      type: Number,
+      default: 0,
+    },
+    manualMode: {
+      type: Boolean,
+      default: false,
+    },
+    openDelay: {
+      type: Number,
+      default: 0,
+    },
+    offset: {
+      type: [Number, Array] as PropType<[number, number] | number>,
+      default: [0, 12] as [number, number],
+      validator: (val: [number, number] | number): boolean => {
+        return isArray(val) && val.length === 2 || typeof val === 'number'
+      },
+    },
+    placement: {
+      type: String as PropType<Placement>,
+      default: 'bottom' as Placement,
+    },
+    popperClass: {
+      type: String,
+      default: '',
+    },
+    referrer: {
+      type: HTMLElement as PropType<Nullable<HTMLElement>>,
+      default: null as Nullable<HTMLElement>,
+    },
+    showArrow: {
+      type: Boolean,
+      default: true,
+    },
+    strategy: {
+      type: String as PropType<PositioningStrategy>,
+      default: 'fixed' as PositioningStrategy,
+    },
+    transition: {
+      type: String,
+      default: 'el-fade-in-linear',
+    },
+    tabIndex: {
+      type: String,
+      default: '0',
+    },
+    value: {
+      type: Boolean,
+      default: true,
+    },
+  },
+  setup(props) {
+    const popperRef = ref(null as Nullable<HTMLElement>)
+    const arrowRef = ref(null as Nullable<HTMLElement>)
+    const popperInstance = ref(null as PopperInstance)
+    const exceptionState = ref(false)
+    const timeout = ref(null as NodeJS.Timeout)
+    const timeoutPending = ref(null as NodeJS.Timeout)
+    const show = ref(false)
+    const tooltipId = ref(`el-tooltip-${generateId()}`)
+
+    const visible = computed(() => {
+      return props.manualMode ? props.value : !props.disabled && show.value
+    })
+
+    const clearTimer = (timer: Ref<Nullable<NodeJS.Timeout>>) => {
+      clearTimeout(timer.value)
+      timer.value = null
+    }
+    // this is a reference that we need to pass down to child component
+    // to obtain the child instance
+    function doDestroy(forceDestroy: boolean) {
+      /* istanbul ignore if */
+      if (!popperInstance.value || (visible.value && !forceDestroy)) return
+      popperInstance.value.destroy()
+      popperInstance.value = null
+      const referrer = props.referrer
+      off(referrer, 'mouseenter', _show)
+      off(referrer, 'mouseleave', _hide)
+      off(referrer, 'focus', handleFocus)
+      off(referrer, 'blur', handleBlur)
+      if (popperRef.value && props.appendToBody) {
+        off(popperRef.value, 'click', stop)
+        document.body.removeChild(popperRef.value)
+      }
+    }
+
+    const showPopper = () => {
+      if (!exceptionState.value || props.manualMode || props.disabled) return
+      clearTimer(timeout)
+      if (props.openDelay === 0) {
+        show.value = true
+      } else {
+        timeout.value = setTimeout(() => {
+          show.value = true
+        }, props.openDelay)
+      }
+
+      if (props.hideAfter > 0) {
```

Comment:
Here is a side case, if we both have openDelay AND hideAfter, or even the value of hideAfter < openDelay, looks like there would be a trouble.

Reply:
Correct, I'll fix that